engine/runtime-parser/src/main/java/org/enso/compiler/core/IR.java

```diff
@@ -0,0 +1,189 @@
+package org.enso.compiler.core;
+
+import org.enso.compiler.core.ir.DiagnosticStorage;
+import org.enso.compiler.core.ir.Expression;
+import org.enso.compiler.core.ir.IdentifiedLocation;
+import org.enso.compiler.core.ir.MetadataStorage;
+import org.enso.syntax.text.Debug;
+import com.oracle.truffle.api.source.Source;
+import scala.Option;
```

Comment:
`scala.Option`. Don't we want to use Java `Optional` or rather just plain `null`? There is certainly going to be a difference in serializing Java `null` and `Optional.empty()`.

Reply:
I didn't want to replace it at this point, yet, since it will be used in plethora of Scala classes.